File: beginner_tutorials/src/talker.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

#include <sstream>
#include <beginner_tutorials/Num.h>

int main(int argc, char **argv) {
    ros::init(argc, argv, "example_talker");

    ros::NodeHandle nh;


    ros::Publisher std_pub = nh.advertise<std_msgs::String>("chatter", 1000);

    ros::Publisher num_pub = nh.advertise<beginner_tutorials::Num>("num", 1000);

    ros::Rate loop_rate(10);

    int count = 0;
    while (ros::ok()) {
        std_msgs::String msg;

        std::stringstream ss;
        ss << "hello ycao " << count;
        msg.data = ss.str();

        ROS_INFO("%s", msg.data.c_str());

        std_pub.publish(msg);

        beginner_tutorials::Num msg_n;
        msg_n.num = rand();

        num_pub.publish(msg_n);

        ros::spinOnce();

        loop_rate.sleep();
        ++ count;
    }




    return 0;
}

File: beginner_tutorials/scripts/talker.py
#! /usr/bin/env python3

import rospy
import random
from std_msgs.msg import String
from beginner_tutorials.msg import Num

def talker():
    rospy.init_node("beg_talker", anonymous=True)
    
    std_pub = rospy.Publisher("chatter", String, queue_size=10)
    num_pub = rospy.Publisher("num", Num, queue_size=10)

    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        msg = String()

        hello_str = "hello ycao %s" %rospy.get_time()
        msg.data = hello_str
        std_pub.publish(msg)

        msg = Num()
        msg.num = random.randint(1, 100)

        num_pub.publish(msg)


        rospy.loginfo(hello_str)

        rate.sleep()


if __name__ == "__main__":
    talker()